addons/sale_purchase/models/sale_order.py

```diff
@@ -248,8 +248,10 @@ def _purchase_service_prepare_line_values(self, purchase_order, quantity=False):
         else:
             product_ctx.update({'partner_id': purchase_order.partner_id.id})
 
+        product = self.product_id.with_context(**product_ctx)
         return {
-            'name': self.product_id.with_context(**product_ctx).display_name,
+            'name': product.display_name + ''.join(['\n', product.description_purchase])
```

Comment:
this should be equivalent:

```suggestion
            'name': product.display_name + '\n' + product.description_purchase
```